scenes/game/at-bat-manager.lua

```diff
@@ -0,0 +1,75 @@
+--------------------------------------------------------------------
+-- AtBatManager is a state machine that manages an individual at bat
+-- between a pitcher and a batter, including calculating the result of
+-- of an at bat
+--------------------------------------------------------------------
+
+local constants = require("scenes.game.constants")
+local config = require("scenes.game.config")
+local AtBatManager = {}
+
+-- Instantiate AtBatManager (constructor)
+function AtBatManager:new(options)
+  local balls = options.balls or 0
+  local strikes = options.strikes or 0
+  local state = constants.NONE
+
+  local atBatManager = {
+    balls = balls,
+    strikes = strikes,
+    state = state
+  }
+
+  setmetatable(atBatManager, self)
+  self.__index = self
+
+  return atBatManager
+end
+
+-- Roll a random result between the player's floor and ceiling
+-- TODO: add additional logic for determining the roll
+local function roll(player)
+  return math.random(player:getSkill():getFloor(), player:getSkill():getCeiling())
+end
+
+-- Run the matchup for a single pitch
+function AtBatManager:throwPitch(pitcher, batter)
+  local pitcherRoll = roll(pitcher)
+  local batterRoll = roll(batter)
+
+  local resultState = self:updateState(pitcherRoll, batterRoll)
+
+  return resultState, pitcherRoll, batterRoll
+end
+
+-- Update state machine for the at bat
+function AtBatManager:updateState(pitcherRoll, batterRoll)
+  local result = batterRoll - pitcherRoll
+
+  -- Positive value means pitcher wins,
+  -- negative value means batter wins,
```

Comment:
Should be the other way around? Shouldn't the player who rolls the bigger number be the winner?

Reply:
oh yah good point, i switched the order of the logic at some point, will change it back